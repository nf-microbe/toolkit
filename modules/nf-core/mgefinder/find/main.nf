process MGEFINDER_FIND {
    tag "${meta.id}"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://carsonjm/mgefinder:0.0.1' :
        'docker://carsonjm/mgefinder:0.0.1' }"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("${prefix}.mgefinder.find")   , emit: find
    path "versions.yml"                                 , emit: versions

    script:
    def args = task.ext.args   ?: ''
    prefix   = task.ext.prefix ?: "${meta.id}"
    """
    mgefinder \\
        find \\
        ${bam[0]} \\
        -o ${prefix}.mgefinder.find \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgefinder: \$(echo \$( mgefinder --version | tail -n 1 | sed "s/MGEfinder version://" ))
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.mgefinder.find

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgefinder: \$(echo \$( mgefinder --version | tail -n 1 | sed "s/MGEfinder version://" ))
    END_VERSIONS
    """
}
