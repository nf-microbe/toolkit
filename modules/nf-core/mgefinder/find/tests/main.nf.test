nextflow_process {

    name "Test Process MGEFINDER_FIND"
    script "../main.nf"
    process "MGEFINDER_FIND"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "mgefinder"
    tag "mgefinder/formatbam"
    tag "mgefinder/find"

    test("fasta.gz - bam") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file("/mmfs1/gscratch/pedslabs_hoffman/carsonjm/nf-microbe/modules/.nf-test/tests/39420309151e6a4c8e2282dfdcef3e60/work/40/23f32dca7ddc95a12cc67208833363/test.mgefinder.bam", checkIfExists: true),
                        file("/mmfs1/gscratch/pedslabs_hoffman/carsonjm/nf-microbe/modules/.nf-test/tests/39420309151e6a4c8e2282dfdcef3e60/work/40/23f32dca7ddc95a12cc67208833363/test.mgefinder.bam.bai", checkIfExists: true)
                    ]
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("bam - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    []
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
