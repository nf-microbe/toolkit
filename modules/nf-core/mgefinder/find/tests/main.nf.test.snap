{
    "bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mgefinder.find:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ed932d2bdd2644829430ce7a5bdb3a12"
                ],
                "find": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mgefinder.find:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ed932d2bdd2644829430ce7a5bdb3a12"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-07T16:06:51.444151494"
    },
    "fasta.gz - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mgefinder.find:md5,a5e34fa7c7cfcb1f121d4ce5e15390be"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ed932d2bdd2644829430ce7a5bdb3a12"
                ],
                "find": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mgefinder.find:md5,a5e34fa7c7cfcb1f121d4ce5e15390be"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ed932d2bdd2644829430ce7a5bdb3a12"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-07T16:26:03.079044035"
    }
}